### PR TITLE
Boolean properties and misc

### PR DESCRIPTION
**Pull request by cataphract**
As mentioned in https://github.com/sensu/sensu-puppet/commit/05999a605345ed49e455d6c92cbb34dc67fd3b85#commitcomment-10864529 , implement the boolean properties through a superclass.

Unfortunately, `Puppet::Parameter::Boolean` was only introduced in 3.3, so we must implement our own superclass. In the process, fixed some other issues.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -30,7 +30,7 @@ end
 if puppetversion = ENV['PUPPET_GEM_VERSION']
   gem 'puppet', puppetversion, :require => false
 else
-  gem 'puppet', :require => false
+  gem 'puppet', '< 4.0.0', :require => false
```
Comment (jamtur01) [inline Gemfile]: Why this?

Comment (cataphract) [inline Gemfile]: puppetlabs_spec_helper-0.10.2 seems to be incompatible with puppet 4.0. I get this error:

```
/usr/bin/ruby1.9.1 -S rspec spec/classes/sensu_api_spec.rb spec/classes/sensu_client_spec.rb spec/classes/sensu_init_spec.rb spec/classes/sensu_package_spec.rb spec/classes/sensu_rabbitmq_spec.rb spec/classes/sensu_redis_spec.rb spec/classes/sensu_server_spec.rb spec/defines/sensu_check_spec.rb spec/defines/sensu_extension_spec.rb spec/defines/sensu_filter_spec.rb spec/defines/sensu_handler_spec.rb spec/defines/sensu_plugin_spec.rb spec/defines/sensu_subscription_spec.rb spec/unit/puppet_x_sensu_totype_spec.rb spec/unit/sensu_config_spec.rb --color
/home/glopes/.gems/gems/puppetlabs_spec_helper-0.10.2/lib/puppetlabs_spec_helper/module_spec_helper.rb:23:in `block in <top (required)>': undefined method `environmentpath=' for #<RSpec::Core::Configuration:0x000000012d2ae8> (NoMethodError)
	from /var/lib/gems/1.9.1/gems/rspec-core-2.99.2/lib/rspec/core.rb:154:in `configure'
	from /home/glopes/.gems/gems/puppetlabs_spec_helper-0.10.2/lib/puppetlabs_spec_helper/module_spec_helper.rb:22:in `<top (required)>'
	from /home/glopes/repos/sensu-puppet/spec/spec_helper.rb:2:in `require'
	from /home/glopes/repos/sensu-puppet/spec/spec_helper.rb:2:in `<top (required)>'
	from /home/glopes/repos/sensu-puppet/spec/classes/sensu_api_spec.rb:1:in `require'
	from /home/glopes/repos/sensu-puppet/spec/classes/sensu_api_spec.rb:1:in `<top (required)>'
	from /var/lib/gems/1.9.1/gems/rspec-core-2.99.2/lib/rspec/core/configuration.rb:1065:in `load'
	from /var/lib/gems/1.9.1/gems/rspec-core-2.99.2/lib/rspec/core/configuration.rb:1065:in `block in load_spec_files'
	from /var/lib/gems/1.9.1/gems/rspec-core-2.99.2/lib/rspec/core/configuration.rb:1065:in `each'
	from /var/lib/gems/1.9.1/gems/rspec-core-2.99.2/lib/rspec/core/configuration.rb:1065:in `load_spec_files'
	from /var/lib/gems/1.9.1/gems/rspec-core-2.99.2/lib/rspec/core/command_line.rb:18:in `run'
	from /var/lib/gems/1.9.1/gems/rspec-core-2.99.2/lib/rspec/core/runner.rb:103:in `run'
	from /var/lib/gems/1.9.1/gems/rspec-core-2.99.2/lib/rspec/core/runner.rb:17:in `block in autorun'
```

Comment (cataphract) [inline Gemfile]: Actually, the odd thing is that line is `c.environmentpath = spec_path if Puppet.version.to_f >= 4.0`, so it's there specifically for puppet 4.

**Comment by jlambert121** [comment]
I've been re-evaluating what versions of puppet I support with my modules and was thinking about this module as well.  The release of 4.0 makes me think we should drop all 2.7 support, and I believe dropping everything pre-3.3 still keeps one version of PE older than current support.  Not to muddy the waters on this, but I'm wondering if it makes sense to just use `Puppet::Parameter::Boolean` and make this module depend on >= 3.3.

**Comment by jamtur01** [comment]
I guess I am not adverse to that - we'll go a major release to go with it?

**Comment by jlambert121** [comment]
That would probably make sense if we remove older supported versions.  If there's other things we want to change with that we should probably have that discussion outside this PR.

**Comment by jamtur01** [comment]
Yeah I don't have anything else - unless you do - there's probably a few types/providers folks would like to replace but I don't have any bandwidth.

**Comment by cataphract** [comment]
So would you prefer a pull request using `Puppet::Property::Boolean` for puppet >= 3.3.0?

**Comment by jamtur01** [comment]
Yes please!

**Comment by cataphract** [comment]
Done. I also added another commit for using the full GPG fingerprint for the apt repository, as this generates a warning in the latest puppetlabs-apt (I can submit a separate PR if you prefer).

**Comment by jamtur01** [comment]
Thanks so much for this - excellent refactor!

**Comment by cataphract** [comment]
Unfortunately, there is a problem. Once a property is true, you cannot change it back to false. This is because on `ResourceHarness::sync_if_needed()` the property value is not even checked to be in sync of if the `should` value is false:

```
if param.should && !param.safe_insync?(current_value)
```

Arguable this should test `nil?` (property unmanaged) but things are what they are. I'll prepare a fix later this week (probably it will have no normalize to `:true`/`:false`), but in the meantime you may want to revert ec9fa40.

**Pull request merged**